src/Uno.UI/UI/Xaml/Controls/MenuFlyout/MenuFlyoutSubItem.cs

```diff
@@ -1,47 +1,889 @@
+			{
+				hasToggleMenuItem = spPresenter.GetContainsToggleItems();
+				hasIconMenuItem = spPresenter.GetContainsIconItems();
+			}
+
+			if (m_tpPopup != null)
+			{
+				bIsPopupOpened = (m_tpPopup.IsOpen);
+			}
+
+			// CommonStates
+			if (!bIsEnabled)
+			{
+				VisualStateManager.GoToState(this, "Disabled", bUseTransitions);
+			}
+			else if (bIsPopupOpened)
+			{
+				VisualStateManager.GoToState(this, "SubMenuOpened", bUseTransitions);
+			}
+			else if (m_menuHelper.IsPressed)
+			{
+				VisualStateManager.GoToState(this, "Pressed", bUseTransitions);
+			}
+			else if (m_menuHelper.IsPointerOver)
+			{
+				VisualStateManager.GoToState(this, "PointerOver", bUseTransitions);
+			}
+			else
+			{
+				VisualStateManager.GoToState(this, "Normal", bUseTransitions);
+			}
+
+			// FocusStates
+			if (FocusState.Unfocused != focusState && bIsEnabled)
+			{
+				if (FocusState.Pointer == focusState)
+				{
+					VisualStateManager.GoToState(this, "PointerFocused", bUseTransitions);
+				}
+				else
+				{
+					VisualStateManager.GoToState(this, "Focused", bUseTransitions);
+				}
+			}
+			else
+			{
+				VisualStateManager.GoToState(this, "Unfocused", bUseTransitions);
+			}
+
+			// CheckPlaceholderStates
+			if (hasToggleMenuItem && hasIconMenuItem)
+			{
+				VisualStateManager.GoToState(this, "CheckAndIconPlaceholder", bUseTransitions);
+			}
+			else if (hasToggleMenuItem)
+			{
+				VisualStateManager.GoToState(this, "CheckPlaceholder", bUseTransitions);
+			}
+			else if (hasIconMenuItem)
+			{
+				VisualStateManager.GoToState(this, "IconPlaceholder", bUseTransitions);
+			}
+			else
+			{
+				VisualStateManager.GoToState(this, "NoPlaceholder", bUseTransitions);
+			}
+
+			// PaddingSizeStates
+			if (shouldBeNarrow)
+			{
+				VisualStateManager.GoToState(this, "NarrowPadding", bUseTransitions);
+			}
+			else
+			{
+				VisualStateManager.GoToState(this, "DefaultPadding", bUseTransitions);
+			}
+
+
+		}
+
+		// MenuFlyoutSubItem's presenter size changed event handler that
+		// adjust the sub presenter position to the proper space area
+		// on the available window rect.
+		void OnPresenterSizeChanged(
+		object pSender,
+		SizeChangedEventArgs args)
+		{
+#if MFSI_DEBUG
+			IGNOREHR(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "MFSI[0x%p]: OnPresenterSizeChanged.", this));
+#endif // MFSI_DEBUG
+
+			m_menuHelper.OnPresenterSizeChanged(pSender, args, m_tpPopup as Popup);
+
+			if (m_tpMenuPopupThemeTransition == null)
+			{
+
+				MenuFlyoutPresenter parentMenuFlyoutPresenter;
+				parentMenuFlyoutPresenter = GetParentMenuFlyoutPresenter();
+
+				// Get how many sub menus deep we are. We need this number to know what kind of Z
+				// offset to use for displaying elevation. The menus aren't parented in the visual
+				// hierarchy so that has to be applied with an additional transform.
+				int depth = 1;
+				if (parentMenuFlyoutPresenter != null)
+				{
+					depth = parentMenuFlyoutPresenter.GetDepth() + 1;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this useless assignment to local variable 'depth'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5205492)